controllers/helper_test.go

```diff
@@ -538,6 +540,52 @@ func testBuildManagedZone(name, ns, domainName string) *kuadrantdnsv1alpha1.Mana
 	}
 }
 
+// testEndpointsTraversable consumes an array of endpoints and returns a boolean
+// indicating presence of that path from host to all IPs
+func testEndpointsTraversable(endpoints []*endpoint.Endpoint, host string, ips []string) bool {
```

Comment:
those are always an array of IP addresses :thinking:  why rename?

Reply:
Why does it have to be limited to checking for ip targets?

Reply:
right now it is "I am looking for IPs. Havent found them but still have where to look / found them so succeed / haven't found them and novhere to look so fail". If I will allow it to look for targets there will be two paths: "I looking for a CNAME. Found it so look for it's target / didn't found it - fail" and "I'm looking for an IPs" one. The difference would be an extra `if` statement that will decide if to say it is traversable or use the current `target` as a new host (the later is done now by just recurring). And this will, actually, lead to more calls (more expecive to run) if I'm not mistaken.